cocos/2d/CCScene.cpp

```diff
@@ -167,7 +172,7 @@ void Scene::render(Renderer* renderer)
         //clear background with max depth
         camera->clearBackground(1.0);
         //visit the scene
-        visit(renderer, transform, 0);
+        visit(renderer, transform, Node::FLAGS_TRANSFORM_DIRTY);
```

Comment:
_HitTest is vritual function in UI::Widget class before, i reused this
interface and move it to Node class.
If Change it to a Helper function like HitTestInNodeContent(...), i'll move
hittest back to UI::Widget._

If hitTest is only used in UI components, then, yes, I think it is better
to move it back to ui::Widget.

Thanks



On 12 June 2015 at 00:41, Vincent Yang <notifications@github.com> wrote:

> In cocos/2d/CCScene.cpp
> <https://github.com/cocos2d/cocos2d-x/pull/12277#discussion_r32295722>:
>
> > @@ -167,7 +172,7 @@ void Scene::render(Renderer* renderer)
> >          //clear background with max depth
> >          camera->clearBackground(1.0);
> >          //visit the scene
> > -        visit(renderer, transform, 0);
> > +        visit(renderer, transform, Node::FLAGS_TRANSFORM_DIRTY);
>
> HitTest is vritual function in UI::Widget class before, i reused this
> interface and move it to Node class.
>
> If Change it to a Helper function like HitTestInNodeContent(...), i'll
> move hittest back to UI::Widget.
> How about this solution?
>
>    - About Node::FLAGS_TRANSFORM_DIRTY problem, i'll try to find out
>    other solution later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cocos2d/cocos2d-x/pull/12277/files#r32295722>.
>